libs/components/config/src/lib/params.ts

```diff
@@ -178,44 +179,67 @@ export class SkyAppRuntimeConfigParams {
 
   /**
    * Adds the current params to the supplied link URL.
+   * @param url The url to update
+   * @param queryParams Optional query parameters to include in the constructed url
    */
-  public getLinkUrl(url: string): string {
+  public getLinkUrl(url: string, queryParams?: Params | null): string {
     return this.#buildUrlWithParams(
       url,
       new Set([
         ...this.#excludeFromRequestsParams,
         ...this.#excludeFromLinksParams,
-      ])
+      ]),
+      queryParams
     );
   }
 
-  #buildUrlWithParams(url: string, excludeParams: Set<string>): string {
-    const existingParams = getUrlSearchParams(url);
+  #buildUrlWithParams(
+    url: string,
+    excludeParams: Set<string>,
+    queryParams?: Params | null
+  ): string {
+    let unifiedParams = getUrlSearchParams(url);
 
-    // When encoding the params, keep the requested params separate so that
-    // we can leave the existing params as-is.
-    let requestedParams = getUrlSearchParams('');
+    // Remove excluded params
+    for (const paramName of excludeParams.keys()) {
+      unifiedParams = unifiedParams.delete(paramName);
+    }
+
+    // Add provided parameters to the URL which are not on the exclusion list
+    // Respect pre-existing params which already have values
+    if (queryParams) {
+      for (const [paramName, decodedValue] of Object.entries(queryParams)) {
+        if (decodedValue) {
```

Comment:
Should `queryParams` respect `excludeParams`? Seems like it should...

Reply:
I'm very much on the fence for this.  I viewed queryParams provided in as the most specific request for arguments to be provided to a link, so I generally felt they should trump the default sky config params or those already on the url.

Do you have a scenario in mind where the general rules for exclusion should trump someone if they explicitly have asked for a query param to be added?  If they didn't want it on there, it feels like they wouldn't be providing it explicitly to this single link.  The url is a bit different, since its already been tampered with by the call chain (adding things like envid).

Reply:
I don't have a specific scenario in mind, and you raise a good argument. We can keep your logic as-is. 👍